File: src/features/describe/describe.prompt.ts
import type { GitContext } from '../../infrastructure/git/git-context.models'

export function buildPrompt(context: GitContext, targets: string[], workItemId: string): string {
  return `## Contexto Git

**Branch:** ${context.branch}
**Base branches alvo:** ${targets.join(', ')}
${workItemId ? `**Work Item:** #${workItemId}\n` : ''}
### Git Log (commits desde a base)

\`\`\`
${context.log}
\`\`\`

### Git Diff

\`\`\`diff
${context.diff}
\`\`\`
`
}

File: lib/src/features/describe/describe_prompt.dart
import '../../domain/change_context.dart';
import 'describe_models.dart';

String buildDescribePrompt(DescribePrompt input) {
  return buildPrompt(input.context, input.targets, input.workItemId);
}

String buildPrompt(
  ChangeContext context,
  List<String> targets,
  String workItemId,
) {
  const fence = '\u0060\u0060\u0060';
  return '''## Contexto Git

**Branch:** ${context.branch}
**Base branches alvo:** ${targets.join(', ')}
${workItemId.isNotEmpty ? '**Work Item:** #$workItemId\n' : ''}### Git Log (commits desde a base)

$fence
${context.log}
$fence

### Git Diff

${fence}diff
${context.diff}
$fence
''';
}
